### PR TITLE
[dagit] avoid duping events on websocket reconnect

### PR DESCRIPTION
**Pull request by alangenfeld**
A scheme in which we move the subscription cursor on websocket reconnection to ensure we dont start loading the events from the begining again

### Test Plan

While loading the events for a large run, stop and start the webserver. Ensure no dupe events are loaded.

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4870** 
    * **PR #4869**  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh) and will continue to be automatically updated while this PR remains open.

File: js_modules/dagit/packages/core/src/runs/LogsProvider.tsx [hunk 1]
```diff
@@ -161,6 +167,20 @@ const useLogsProviderWithSubscription = (runId: string) => {
     },
   });
 
+  // If the websocket reconnects, ensure the newly recreated subscription resumes from
+  // after the messages we have already loaded
+  React.useEffect(() => {
+    if (!websocketClient) {
+      return;
+    }
+    const unlisten = websocketClient.onReconnected(() => {
```
Comment (prha) [inline js_modules/dagit/packages/core/src/runs/LogsProvider.tsx:176]: should this be `onReconnecting` (or `onDisconnected`)?  just not obvious what the lifecycle is between the subscription request and the websocket client.

**Review by prha — APPROVED** [review]

File: js_modules/dagit/packages/core/src/runs/LogsProvider.tsx (hunk 1 above)
Comment (alangenfeld) [reply js_modules/dagit/packages/core/src/runs/LogsProvider.tsx:176]: ya good call - worth experimenting with that

File: js_modules/dagit/packages/core/src/app/useWebsocketAvailability.tsx [hunk 2]
```diff
@@ -3,23 +3,37 @@ import * as React from 'react';
...
+        connUnlisten();
+      };
+    }
 
-    return () => ws.close();
-  }, [websocketURI]);
+    return () => {
+      cleanup();
+    };
+  }, [websocketClient]);
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/app/useWebsocketAvailability.tsx:36]: I don't think we really need to make this one use the Apollo `WebSocketClient`. Its only purpose is to check whether WebSockets are available at all, after which point we don't use it again. So a vanilla `WebSocket` is all we should need.

File: js_modules/dagit/packages/core/src/assets/AssetDetails.tsx [hunk 3]
```diff
@@ -23,9 +23,10 @@ import {AssetQuery, AssetQueryVariables} from './types/AssetQuery';
 interface Props {
   assetKey: AssetKey;
   asOf: string | null;
+  asSidebarSection?: boolean;
 }
 
-export const AssetDetails: React.FC<Props> = ({assetKey, asOf}) => {
+export const AssetDetails: React.FC<Props> = ({assetKey, asOf, asSidebarSection}) => {
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AssetDetails.tsx:29]: Can this stuff be done in a separate PR? I feel like major WS changes belong in a changeset separate from other product changes.

**Review by hellendag — CHANGES_REQUESTED** [review]
Would prefer to have WebSocket changes separate from the other product changes, unless I'm missing a reason for the changes to be combined.

**Comment by hellendag** [comment]
I believe this can be closed in favor of #4915, now that much of it has been merged in #4870.
